subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_pb_gatt_srv);
 
 #define FAST_ADV_TIME (60LL * MSEC_PER_SEC)
 
+#if defined(CONFIG_BT_DEVICE_NAME_DYNAMIC)
+#define BT_DEVICE_NAME_LEN strlen(bt_get_name())
+#else
+#define BT_DEVICE_NAME_LEN (sizeof(CONFIG_BT_DEVICE_NAME) - 1)
+#endif
```

Comment:
Did you consider putting variants of a `#define BT_DEVICE_NAME` into the conditional branches too, so that in the static case you don't end up resolving to a function call but just a string literal? I suspect the compiler will likely optimize this anyway since it should know that the function just returns the string literal, but since you have these conditional branches anyway you could remove some doubt by just having that extra define.